### ISSUE TITLE
Change Request for GET /character/{character_id}/mail OR POST /universe/names/

### ISSUE THREAD

**Issue by ddouglas493** [ccpgames/esi-issues#312]
Current there is no way to determine the type of id that the sender is. I was made aware earlier this morning that it is possible for a Mail to be FROM a mailing list. It a mail could be from any other entity, it had to be a mailing list......UGH. Anyway, the POST /universe/names/ does not accept Mailing List IDs or this would not be an issue.

My Request is this. Please change the from field in the Mail Header to an array listing the id and type like with recipients or Post /universe/names/ to parse Mailing List IDs.

**Comment by ddouglas493** [comment]
This is partially a duplicate of #266. I created a new issue because this offer two different workarounds to fix the issue.

**Comment by ddouglas493** [comment]
EDIT: Grammar

**Comment by ddavaham** [comment]
So with @aquarhead response to #266 I want to update this request to only be to expose that it is a From field be converted to an array which list the ID and ID Type like with the recipients field. No need for this request to be for the POST /universe/names/ endpoint anymore. If I know that it is a mailing list, then I can just look for it on the Mailing List endpoint and if it is there, great, if not, then oh well.

**Comment by ddavaham** [comment]
EDIT: Grammer

**Comment by a-tal** [comment]
the monolith action needs an update to support this, ideally a `sender_type` attribute

**Comment by ddavaham** [comment]
@a-tal How hard would it be to have `/mail/lists` return all mailing lists with a subscribed flag instead of changing the from field?

**Comment by warlof** [comment]
I agree with the initial `from_type`  since we need to use name resolver in order to convert ID to Name and this endpoint is bouncing any request which is containing an invalid ID in the array.

Plus, it's making the endpoint structure similiar to the recipients one.

**Comment by minlexx** [comment]
Currently `/characters/{}/mail/lists/` does not return mailing lists IDs which char is not subscribed to, that's why it is **impossible** in some situations to guess a sender type, and resolve sender or receiver name, because you don't know what type that ID is. My program sees, that it is not in character mailing lists, assumes then the ID is for character, goes to `/universe/names/` to resolve name and it returns just
```
eveapi: 404 "POST" "https://esi.evetech.net/latest/universe/names/"
eveapi: POST data:  "[145174036]"
eveapi:  returned JSON error was: "Ensure all IDs are valid before resolving."
```
I get ++errorCount and even cannot write "Unknown mailing list" in UI. Just empty string :cry:

**Comment by cvweiss** [comment]
Adding the mailing list name to universe/names is the simplest solution IMO, we already have the ID, we only need the name.

**Comment by warlof** [comment]
both solutions sounds valid to me :

1) either add a new field to the mail endpoint called `sender_type` which is matching the pattern to the recipients field

2) or allow mailing list ID resolving from `universe/name` endpoint

In both case we will be able to apply some filter on mail received from a mailing list
Even if the second case sounds more flexible as it also allow us to resolve the mailing list name - assuming it's not a critical information.
